src/parser/expression.js

```diff
@@ -436,18 +436,17 @@ pp.parseExprAtom = function (refShorthandDefaultPos) {
       return id;
 
     case tt._do:
-      if (this.hasPlugin("doExpressions")) {
-        let node = this.startNode();
-        this.next();
-        let oldInFunction = this.state.inFunction;
-        let oldLabels = this.state.labels;
-        this.state.labels = [];
-        this.state.inFunction = false;
-        node.body = this.parseBlock(false, true);
-        this.state.inFunction = oldInFunction;
-        this.state.labels = oldLabels;
-        return this.finishNode(node, "DoExpression");
-      }
```

Comment:
We had a fall-through here before without `doExpression`, but that looks like a mistake anyway. Why should do fall-through to regexp?